app/controllers/api/functions.php

```diff
@@ -610,7 +610,8 @@
     ->inject('project')
     ->inject('deviceFunctions')
     ->inject('deviceLocal')
-    ->action(function (string $functionId, string $entrypoint, mixed $code, bool $activate, Request $request, Response $response, Database $dbForProject, Event $events, Document $project, Device $deviceFunctions, Device $deviceLocal) {
+    ->action(function (string $functionId, string $entrypoint, mixed $code, mixed $activate, Request $request, Response $response, Database $dbForProject, Event $events, Document $project, Device $deviceFunctions, Device $deviceLocal) {
+        $activate = in_array($activate, ['1', 'true', 1, true], true);
```

Comment:
The boolean validator should only check for true and false. Not 1 and 0.
And the loose flag in the validator should also allow the strings 'true' and 'false'

Reply:
So lets update the Boolean Validator